internal/clients/team5/definitions.go

```diff
@@ -0,0 +1,170 @@
+// type GiftRequest shared.Resources
+
+// // GiftRequestDict contains the details of an island's gift requests to everyone else.
+// type GiftRequestDict map[shared.ClientID]GiftRequest
+
+// AcceptReason (Yes/No)
+type AcceptReason int
+
+// GiftResponse is a struct of the response and reason
+type GiftResponse struct {
+	AcceptedAmount shared.Resources
+	Reason         AcceptReason
+}
+
+// GiftInfo holds information about the gifts
+type GiftInfo struct {
+	requested shared.GiftRequest
+	gifted    shared.GiftOffer
+	reason    shared.AcceptReason
+}
+
+//GiftExchange Looks at the exchanges our island has made (Gifts)
+type GiftExchange struct {
+	IslandRequest map[uint]GiftInfo
+	OurRequest    map[uint]GiftInfo
+}
+
+//GiftHistory is the history of our gifts
+type GiftHistory map[shared.ClientID]GiftExchange
+
+/*================================================================
+	Client Information
+=================================================================*/
+type clientConfig struct {
+	// Initial non planned foraging
+	InitialForageTurns uint
+
+	// Skip forage for x amount of returns if theres no return > 1* multiplier
+	SkipForage uint
+
+	// If resources go above this limit we are balling with money
+	JBThreshold shared.Resources
+
+	// Middle class:  Middle < Jeff bezos
+	MiddleThreshold shared.Resources
+
+	// Poor: Imperial student < Middle
+	ImperialThreshold shared.Resources
+
+	// How much to request when we are dying
+	DyingGiftRequest shared.Resources
+
+	// How much to request when we are at Imperial
+	ImperialGiftRequest shared.Resources
+
+	// How much to request when we are dying
+	MiddleGiftRequest shared.Resources
+}
+
+// Client is the island number
+type client struct {
+	*baseclient.BaseClient
+
+	resourceHistory ResourceHistory
+
+	forageHistory ForageHistory // Stores our previous foraging data
+	// giftHistory   GiftHistory
+
+	giftHistory GiftHistory
+
+	taxAmount shared.Resources
+
+	// allocation is the president's response to your last common pool resource request
+	allocation shared.Resources
+
+	config clientConfig
+}
+
+/*================================================================
+	Constants
+=================================================================*/
+// Wealth Tiers
+const (
+	Dying           WealthTier = iota // Sets values = 0
+	ImperialStudent                   // iota sets the folloing values =1
+	MiddleClass                       // = 2
+	JeffBezos                         // = 3
+)
+
+const (
+	// Accept ...
+	Accept AcceptReason = iota
+	// DeclineDontNeed ...
+	DeclineDontNeed
+	// DeclineDontLikeYou ...
+	DeclineDontLikeYou
+	// Ignored ...
+	Ignored
+)
+
+/*================================================================
+	Functions
+=================================================================*/
+func (wt WealthTier) String() string {
+	strings := [...]string{"Dying", "ImperialStudent", "MiddleClass", "JeffBezos"}
```

Comment:
Remember to change these too when you change the other iota definitions to `dying`, `imperialStudent` etc